IoTDeviceSimulator/web-client/src/App.js

```diff
@@ -169,56 +170,63 @@ function App() {
     }
   })
 
-  const openDeviceTwinModal = async (deviceAction, deviceTwin, isButtonView) => {
-    setDeviceTwin({ ...deviceTwin, deviceAction });
+  const openAddTelemetryPointModal = async (deviceAction, telemetryPoint, isButtonView) => {
+    setTelemetryPoint({ ...telemetryPoint, deviceAction });
```

Comment:
`setTelemetryPoint({ ...telemetryPoint, deviceAction })` : this code can lead to unpredictable behavior. add anonymous function to update a state like `setTelemetryPoint((telemetryPoint)=>({ ...telemetryPoint, deviceAction }))` .
Look for similar changes in other places too. 
If state contains object or array, it is always recommended that we update it with anonymous function.